Events.php

```diff
@@ -63,22 +63,18 @@ public static function onServiceWorkerControllerInit($event): void
         // Service Worker Addons
         $controller->additionalJs .= <<<JS
             // Give the service worker access to Firebase Messaging.
-            importScripts('{$bundle->baseUrl}/firebase-app.js');
-            importScripts('{$bundle->baseUrl}/firebase-messaging.js');
-            //importScripts('https://www.gstatic.com/firebasejs/6.3.3/firebase-app.js');
-            //importScripts('https://www.gstatic.com/firebasejs/6.3.3/firebase-messaging.js');
-        
-           firebase.initializeApp({messagingSenderId: "{$pushDriver->getSenderId()}"});
-            
-            const messaging = firebase.messaging();
-            messaging.setBackgroundMessageHandler(function(payload) {
-              const notificationTitle = payload.data.title;
-              const notificationOptions = {
-                body: payload.data.body,
-                icon: payload.data.icon
-              };
-              return self.registration.showNotification(notificationTitle, notificationOptions);
+            importScripts('{$bundle->baseUrl}/firebase-app-compat.js');
+            importScripts('{$bundle->baseUrl}/firebase-messaging-compat.js');
+
+            firebase.initializeApp({
+                messagingSenderId: "{$pushDriver->getSenderId()}",
+                projectId: "{$module->getConfigureForm()->getJsonParam('project_id')}",
+                appId: "{$module->getConfigureForm()->firebaseAppId}",
+                apiKey: "{$module->getConfigureForm()->firebaseApiKey}",
             });
+
+            // Initialize Firebase Cloud Messaging and get a reference to the service
+            const messaging = firebase.messaging();
```

Comment:
@luke- Why do we need `const messaging = ` here, if it is not used below?
Maybe only `firebase.messaging();` is enough?

Reply:
@yurabakhtin Yes, that should be enough. I've copied it from the official doc.
https://firebase.google.com/docs/cloud-messaging/js/client

I'll remove the const